ignition-server/main.go

```diff
@@ -123,10 +126,15 @@ func payloadStoreReconciler(ctx context.Context) error {
 			Namespace: os.Getenv(namespaceEnvVariableName),
 		},
 	}).SetupWithManager(ctx, mgr); err != nil {
+		log.Error(err, "Unable to create controller")
 		return fmt.Errorf("unable to create controller: %w", err)
 	}
 
-	return mgr.Start(ctx)
+	if err := mgr.Start(ctx); err != nil {
+		log.Error(err, "Failed to start manager")
+		return fmt.Errorf("Failed to start manager: %w", err)
+	}
+	return nil
```

Comment:
If this function returns an error, it seems like the program becomes defunct but lives on as a zombie: https://github.com/openshift/hypershift/blob/82091eac14fd0e635cb4494a3e67ae7996aef29a/ignition-server/main.go#L145

Shouldn't the setup code here return an error which is fatal to the program, and only spin up the `Start()` function in a goroutine if setup succeeds?